src/panel/pages/timeline/components/TimelineEvent.tsx

```diff
@@ -1,15 +1,60 @@
-import React, { FC, useContext, useMemo } from "react";
-import styled, { ThemeContext } from "styled-components";
+import React, { FC, useMemo } from "react";
+import styled, {
+  css,
+  FlattenSimpleInterpolation,
+  FlattenInterpolation,
+  ThemeProps,
+  DefaultTheme
+} from "styled-components";
 import { ReceivedDebugEvent } from "../../../types";
 import { useTooltip, TimelineTooltip } from "./TimelineTooltip";
 
-const EventDot = styled.div`
-  background: ${props => props.color};
-  border-radius: 50%;
-  border: solid 2px ${props => props.theme.dark["+1"]};
+const EVENT_SIZE = "10px";
+const shapeMap: Record<
+  string,
+  | FlattenSimpleInterpolation
+  | FlattenInterpolation<ThemeProps<DefaultTheme>>
+  | undefined
+> = {
+  addition: css`
+    border-radius: 50%;
+  `,
+  update: css`
+    border-radius: 0%;
+  `,
+  teardown: css`
+    position: relative;
+    background-color: transparent;
+    &:before,
+    &:after {
+      content: " ";
+      position: absolute;
+      height: 100%;
+      width: calc(${EVENT_SIZE} / 3);
+    }
+    &:before {
+      background-color: ${p => p.theme.grey["+3"]};
+      transform: translateX(100%) rotate(45deg);
+    }
+    &:after {
+      background-color: ${p => p.theme.grey["+3"]};
+      transform: translateX(100%) rotate(-45deg);
+    }
+  `
+};
```

Comment:
Doing this should give us a little more type safety and remove the need to define the record type

```suggestion
} as const;
```